src/test/java/com/example/demo/controller/BooksControllerTest.java

```diff
@@ -60,4 +61,14 @@ public void searchTest() throws Exception {
                         .isOk())
                 .andExpect(content().string(containsString("1")));
     }
+
+    @Test
+    public void filterTest() throws Exception {
+        when(service.findBooksWithParams(any())).thenReturn(result);
+        mockMvc.perform(get("/filter"))
+                .andDo(print())
```

Comment:
`.andDo(print())` to nie powinno być w kodzie poza okresem, w którym naprawiasz test ;) Do usunięcia

Reply:
done